src/dart_diagnostic_provider.ts

```diff
@@ -48,4 +51,12 @@ export class DartDiagnosticProvider {
 				throw new Error("Unknown severity type: " + severity); 
 		}
 	}
+
+	private flushResults(notification: as.AnalysisFlushResultsNotification) {		
+		if (notification.files) {
```

Comment:
Do we need this `if`? The docs don't say this is optional and it seems unlikely this event would be raised without any files?